NextGenConnect/JSON to HL7v2.xml

```diff
@@ -83,35 +83,257 @@ for (var i = 0; i &lt; getArrayOrXmlLength(patient[&apos;identifier&apos;]); i++
+tmp[&apos;PID&apos;][&apos;PID.5&apos;][&apos;PID.5.7&apos;] = &quot;L&quot;;  // Legal Name		
+
+//PID-6 Mothers Maiden Name -- ??? Try to grab value ?
+
+//PID-7 Birthdate 
+
+var birthDate = patient[&apos;birthDate&apos;];
+
+birthDate.replace(&apos;-&apos;,&apos;&apos;); // Replaces -&apos;s in birthDate
+
+tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = birthDate;
+
+
+//PID-8 Administrative Sex -- Needs to be edited for actualy values F/M
+
+if(patient[&apos;gender&apos;] == &apos;female&apos;){  
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;F&apos;;
+}
+else if(patient[&apos;gender&apos;] == &quot;male&quot;){
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;M&apos;;
+	}
+
+else{
+	tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = &apos;U&apos;;
+}
+
+//PID-10 Race  -- ??? Try to grab value ?
+
 
-tmp[&apos;PID&apos;][&apos;PID.7&apos;][&apos;PID.7.1&apos;] = patient[&apos;birthDate&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.8&apos;][&apos;PID.8.1&apos;] = patient[&apos;gender&apos;][&apos;@value&apos;].toString();
 
 //PID-11 address
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][&apos;line&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.7&apos;] = patient[&apos;address&apos;][&apos;use&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.3&apos;] = patient[&apos;address&apos;][&apos;city&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.4&apos;] = patient[&apos;address&apos;][&apos;state&apos;][&apos;@value&apos;].toString();
-tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.5&apos;] = patient[&apos;address&apos;][&apos;postalCode&apos;][&apos;@value&apos;].toString();
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.1&apos;] = patient[&apos;address&apos;][0][&apos;line&apos;][0];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.3&apos;] = patient[&apos;address&apos;][0][&apos;city&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.4&apos;] = patient[&apos;address&apos;][0][&apos;state&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.5&apos;] = patient[&apos;address&apos;][0][&apos;postalCode&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.6&apos;] = patient[&apos;address&apos;][0][&apos;country&apos;];
+tmp[&apos;PID&apos;][&apos;PID.11&apos;][&apos;PID.11.7&apos;] = &apos;M&apos; // Mailing address
 
-//PID-13 Home Phone Number
-tmp[&apos;PID&apos;][&apos;PID.13&apos;][&apos;PID.13.1&apos;] = patient[&apos;telecom&apos;][0][&apos;value&apos;][&apos;@value&apos;].toString();
+//PID-13 Home Phone Number -- Need to seperate the area code and number if sending...
+tmp[&apos;PID&apos;][&apos;PID.13&apos;][&apos;PID.13.1&apos;] = patient[&apos;telecom&apos;][0][&apos;value&apos;];
```

Comment:
You'll need to parse the area code out or WIR won't take it. Try something like this: https://stackoverflow.com/questions/13868130/parse-area-code-out-of-phone-number

Reply:
Thanks for the link. Will sort this out.

Reply:
Done.